embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpanImpl.kt

```diff
@@ -114,24 +119,37 @@ internal class EmbraceSpanImpl(
         }
     }
 
-    override fun addEvent(name: String, timestampMs: Long?, attributes: Map<String, String>?): Boolean {
-        if (eventCount.get() < MAX_EVENT_COUNT && inputsValid(name, attributes)) {
-            val newEvent = EmbraceSpanEvent.create(
+    override fun addEvent(name: String, timestampMs: Long?, attributes: Map<String, String>?): Boolean =
+        recordEvent(name = name, attributes = attributes) {
+            EmbraceSpanEvent.create(
                 name = name,
                 timestampMs = timestampMs?.normalizeTimestampAsMillis() ?: openTelemetryClock.now().nanosToMillis(),
                 attributes = attributes
             )
-            synchronized(eventCount) {
-                if (eventCount.get() < MAX_EVENT_COUNT && isRecording) {
-                    events.add(newEvent)
-                    eventCount.incrementAndGet()
-                    return true
-                }
-            }
         }
 
-        return false
-    }
+    override fun recordException(exception: Throwable, attributes: Map<String, String>?): Boolean =
```

Comment:
Mirror the behaviour of the Java SDK. I can't simply defer to it because we need to support snapshotting an unfinished span, so I do the translation here. Doing it here also has the advantage of being to extend this functionality by allowing us to specify a specific time for the exception, which can be added later.